contrib/intel/jenkins/common.py

```diff
@@ -23,37 +23,29 @@ def run_command(command):
         sys.exit(p.returncode)
 
 
-Prov = collections.namedtuple('Prov', 'core util')
+Prov = collections.namedtuple('Prov', 'core enable force_dl')
 prov_list = [
+    Prov("psm2", True, True), # psm2 is build with psm2 src as dl provider
```

Comment:
why is the psm2 build forced to dl?

Reply:
The build modes can be dl, dbg or regular( no enable-debug flag). These arguments are supplied from the Jenkinsfile. The enable forced dl will cause the enable-psm2 to be set to dl even though the requested was a debug or a regular build. 
...
./configure --prefix=/xxx/PR-6560/7/dbg **--enable-debug --enable-psm2=dl** --enable-psm3=yes --enable-sockets=yes -- ....

Reply:
The build settings should control this.  The script should not override the configure option itself by forcing dl build when that was not requested.  Please drop this part of the change.

Reply:
Currently psm3 has the limitation that when both psm2 and psm3 are built as static providers and psm2 is built with libpsm2 embeded (with-psm2-src), there are linking issues. See #6575 

I choose to change psm2 here as the configuration option, while supported is non-standard. I can force_dl to psm3 while we fix thsi issue in psm3 code? which ever you prefer.

Reply:
We need to avoid introducing a change that can be viewed as a regression. #6575 looks like the correct short term solution.

Reply:
#6580 is the proposed fix to allow both to be built as static libs.